src/decoder/zlib.rs

```diff
@@ -101,10 +111,9 @@ impl ZlibStream {
             return Ok(());
         }
 
-        loop {
+        while !self.state.is_done() {
             self.prepare_vec_for_appending();
-
-            let (in_consumed, out_consumed) = self
```

Comment:
`in_consumed` is always 0, since (after removal of `in_buffer`) we always pass an empty slice `&[]` to the `read` method on old line 109 /  new line 118.